generator/explores/explore.py

```diff
@@ -95,6 +95,55 @@ def get_view_lookml(self, view: str) -> dict:
             return lkml.load((self.views_path / f"{view}.view.lkml").read_text())
         raise Exception("Missing view path for get_view_lookml")
 
+    def get_unnested_fields_joins_lookml(
+        self,
+    ) -> list:
+        """Get the LookML for joining unnested fields."""
+        views_lookml = self.get_view_lookml(self.views["base_view"])
+        views: List[str] = [view["name"] for view in views_lookml["views"]]
+        parent_base_name = views_lookml["views"][0]["name"]
+
+        extended_views: List[str] = []
+        if "extended_view" in self.views:
+            # check for extended views
+            extended_views_lookml = self.get_view_lookml(self.views["extended_view"])
+            extended_views = [view["name"] for view in extended_views_lookml["views"]]
+
+            views_lookml.update(extended_views_lookml)
+            views += extended_views
+
+        joins = []
+        for view in views_lookml["views"][1:]:
+            view_name = view["name"]
+            # get repeated, nested fields that exist as separate views in lookml
+            base_name, metric = self._get_base_name_and_metric(
+                view_name=view_name, views=views
+            )
+            metric_name = view_name
+            metric_label = metric_name.replace("_", " ").title()
+
+            if view_name in extended_views:
+                # names of extended views are overriden by the name of the view that is extending them
+                metric_label = (
+                    metric_name.replace(base_name, parent_base_name)
+                    .replace("_", " ")
+                    .title()
```

Comment:
Seeing this pattern twice (and I'll bet there's more of it elsewhere in the repository). Time to make a utility function? Can be done as a followup of course.

Reply:
Turns out we already have a `slug_to_title` utility function